pkg/controller/data_upload_controller.go

```diff
@@ -399,16 +415,12 @@ func (r *DataUploadReconciler) SetupWithManager(mgr ctrl.Manager) error {
 
 func (r *DataUploadReconciler) findDataUploadForPod(podObj client.Object) []reconcile.Request {
 	pod := podObj.(*corev1.Pod)
-
-	du := &velerov2alpha1api.DataUpload{}
-	err := r.client.Get(context.Background(), types.NamespacedName{
-		Namespace: pod.Namespace,
-		Name:      pod.Labels[velerov1api.DataUploadLabel],
-	}, du)
-
+	du, err := findDataUploadByPod(r.client, *pod)
 	if err != nil {
 		r.logger.WithField("Backup pod", pod.Name).WithError(err).Error("unable to get dataupload")
 		return []reconcile.Request{}
+	} else if du == nil {
```

Comment:
The same as above, add a log or assertion

Reply:
Added